src/state/trajectory/logics.ts

```diff
@@ -345,6 +353,104 @@ const loadFileViaUrl = createLogic({
     type: LOAD_FILE_VIA_URL,
 });
 
+// configures the controller for file conversion and sends server health checks
+const configureFileConversionLogic = createLogic({
+    process(
+        deps: ReduxLogicDeps,
+        dispatch: <T extends AnyAction>(action: T) => T,
+        done
+    ) {
+        const { getState } = deps;
+
+        // TODO: Most likely this will eventually replace the config up top
+        // but for development purposes it's here
+        // until we switch to Octopus, make sure it matches your local instance.
+        const netConnectionConfig: NetConnectionParams = {
+            serverIp: "0.0.0.0",
+            serverPort: 8765,
+            useOctopus: true,
+            secureConnection: false,
+        };
+        // check if a controller exists and has the right configuration
+        // create/configure as needed and put in state
+        let controller = getSimulariumController(getState());
+        if (!controller) {
+            dispatch(
+                setConversionStatus({
+                    status: CONVERSION_NO_SERVER,
+                })
+            );
+            controller = new SimulariumController({
+                netConnectionSettings: netConnectionConfig,
+            });
+            dispatch(setSimulariumController(controller));
+        } else if (!controller.remoteWebsocketClient) {
+            controller.configureNetwork(netConnectionConfig);
+        }
+        // check the server health
+        // Originally thought to send checks every 15 seconds,
+        // currently sending a flurry of checks, 3 seconds apart, 15 seconds total.
+        // If any come back true we assume we're good for now... this timing is arbitrary
+        // I reasoned a quick set of requests that can be started on page load,
+        // that cumulatively have enough delay to let a connection be configured
+        // is a better idea than a minute long period of sending checks,
+        // during which things could change.
+        // We can retrigger this as often as necessary.
```

Comment:
This is a good explanation, thank you! Maybe worth shortening this comment in-code though, since this seems like a comment for GitHub/PR

Reply:
Indeed! I'll put a TODO flag at the top to make sure to cut it down before merging.